python/indexing/regex_mlir_bindings.py

```diff
@@ -0,0 +1,61 @@
+import re
+import sys
+from os.path import exists
+
+symbol_guard = """\
+#ifdef MY_PYBIND11_EXPORT_SYMBOLS
+#  define MY_PYBIND11_EXPORT __attribute__((visibility("default")))
+#else
+#  define MY_PYBIND11_EXPORT
+#endif
+"""
+
+module_guard = """\
+#ifdef MY_PYBIND11_EXPORT_MODULE
+#  define MY_PYBIND11_LOCAL_MODULE
+#else
+#  define MY_PYBIND11_LOCAL_MODULE , py::module_local()
+#endif
+"""
```

Comment:
I'm not sure if this is working correctly (will test further) but the idea is to prevent Iterators from taking on the same hacks as Indexing (currently) does.